### ISSUE TITLE
Octobercms with PostgreSQL DB showing error in blog page

### ISSUE THREAD

**Issue by akhileshkcoder** [octobercms/october#5444]
I have installed 'octobercms' using composer. I have installed the theme MegaKit, now the problem is that when I open the blog page in the front-end it is showing an error.

![image](https://user-images.githubusercontent.com/62323833/103532294-febde300-4eb0-11eb-98d0-05138155ce7d.png)

I am using PostgreSQL as the database. The problem seems to be with the 'rainlab' plugin. I am unable to debug the code. Please help.

**Comment by LukeTowers** [comment]
Please post the full stack trace.

**Comment by akhileshkcoder** [comment]
@LukeTowers 

![image](https://user-images.githubusercontent.com/62323833/103610983-51e27500-4f47-11eb-8ca9-70a2c9674577.png)

**Comment by bennothommo** [comment]
@akhileshkcoder @LukeTowers so it appears this issue is caused by the fact we are comparing a numeric ID against the `attachment_id` column in the `system_files` table, which is a varchar string column. PostgreSQL does not like this, it seems.

Given that we can't change `attachment_id` to an integer, as there's the possibility for string keys, we'll have to investigate a way to make the comparison string-based for PostgreSQL only.

**Comment by JamesInform** [comment]
You can create an implizit cast for PostgreSQL.

Just run the following Statement once on the PostgreSQL Database.

`CREATE CAST (varchar AS integer) WITH INOUT AS IMPLICIT; `

This will instruct to automatically convert a Varchar to Integer if used in a comparison. That should do the trick.

See also: [PostgreSQL Documentation - Create Cast](https://www.postgresql.org/docs/current/sql-createcast.html)

Cheers,
James

**Comment by LukeTowers** [comment]
@bennothommo isn't this related to https://github.com/octobercms/library/pull/502?

**Comment by RomainMazB** [comment]
It definitely is, I can confirm!

The problem occurs when an we eager load File, here the blogPosts component eager load the featured images, but it also occurs on Forum plugin with the avatars eager loaded on topic page, and everywhere else where eager load is wanted with the File model.

For my projects I migrated to MySQL waiting for the PR to be merged :)

Edit: and after viewing the @bennothommo and @JamesInform proposals, the way I've fixed it sounds to be the most appropriate way.
The problem has been introduced with the L6 update, this is due to an update into Laravel 5.7, [details here](https://github.com/octobercms/october/issues/5200#issuecomment-655758684).
